File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/did/GetDidMetadataTest.java
package org.hyperledger.indy.sdk.did;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.wallet.WalletValueNotFoundException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class GetDidMetadataTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testGetDidMetadataWorks() throws Exception {
		Did.setDidMetadata(wallet, DID, METADATA).get();
		String receivedMetadata = Did.getDidMetadata(wallet, DID).get();
		assertEquals(METADATA, receivedMetadata);
	}

	@Test
	public void testGetDidMetadataWorksForEmptyString() throws Exception {
		Did.setDidMetadata(wallet, DID, "").get();
		String receivedMetadata = Did.getDidMetadata(wallet, DID).get();
		assertEquals("", receivedMetadata);
	}

	@Test
	public void testGetDidMetadataWorksForNoMetadata() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletValueNotFoundException.class));

		Did.getDidMetadata(wallet, DID).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/DidTests/GetDidMetadataTest.cs
using Hyperledger.Indy.DidApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.DidTests
{
    [TestClass]
    public class GetDidMetadataTest : IndyIntegrationTestWithSingleWallet
    {
        [TestMethod]
        public async Task TestGetDidMetadataWorks()
        {
            await Did.SetDidMetadataAsync(wallet, DID, METADATA);
            var receivedMetadata = await Did.GetDidMetadataAsync(wallet, DID);

            Assert.AreEqual(METADATA, receivedMetadata);
        }

        [TestMethod]
        public async Task TestGetDidMetadataWorksForEmptyString()
        {
            await Did.SetDidMetadataAsync(wallet, DID, string.Empty);
            var receivedMetadata = await Did.GetDidMetadataAsync(wallet, DID);

            Assert.AreEqual(string.Empty, receivedMetadata);
        }

        [TestMethod]
        public async Task TestGetDidMetadataWorksForNoMetadata()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletItemNotFoundException>(() =>
               Did.GetDidMetadataAsync(wallet, DID)
           );
        }
    }
}